### PR TITLE
Reuse existing editor keep history

### PR DESCRIPTION
**Pull request by adiux**
Add additional options for the capabilities and behavior of the grapesjs builder. 

This is a companion PR for the changes in the mautic/mautic repo. https://github.com/mautic/mautic/pull/10781

These changes have already been used to build it. So it should be enough to test that PR in the mautic/mautic repo. For this preset PR, a code review should be enough.

### CODE REVIEW

**Comment by RCheesley** [comment]
I think we still have an issue with the PR in mautic/mautic so will hold off this one until it's resolved.

**Comment by RCheesley** [comment]
@adiux is this PR still valid, if so I think we need to recreate https://github.com/mautic/mautic/pull/10847?

File: src/buttons.js [hunk 1]
```diff
@@ -108,45 +109,52 @@ export default (editor, opts = {}) => {
...
     // Load and show settings and style manager
-    const openTmBtn = pm.getButton('views', 'open-tm');
-    const openSm = pm.getButton('views', 'open-sm');
-    if (openTmBtn) {
-      openTmBtn.set('active', 1);
-    }
-    if (openSm) {
-      openSm.set('active', 1);
-    }
+    if (!opts.combineSettingsAndSm) {
```
Comment (adiux) [inline src/buttons.js:112]: this part is merged in our repo

File: src/buttons.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import ButtonClose from './buttons/buttonClose';
 export default (editor, opts = {}) => {
   const { $ } = editor;
   const pm = editor.Panels;
+  const defaultPanel = opts.defaultPanel || 'open-blocks';
```
Comment (adiux) [inline src/buttons.js:6]: this part is merged and live in our repo

File: src/buttons.js [hunk 3]
```diff
@@ -108,45 +109,52 @@ export default (editor, opts = {}) => {
...
+        }
+      });
+
+      // Open settings
+      traitsProps.get(0).style.display = 'block';
+    }
 
-    // Open block manager
-    const openBlocksBtn = editor.Panels.getButton('views', 'open-blocks');
+    // Open the default panel
```
Comment (adiux) [inline src/buttons.js:149]: this part is merged and live

File: src/buttons.js [hunk 4]
```diff
@@ -108,45 +109,52 @@ export default (editor, opts = {}) => {
...
+    const openBlocksBtn = editor.Panels.getButton('views', defaultPanel);
     if (openBlocksBtn) {
       openBlocksBtn.set('active', 1);
     }
   });
+  // Workaround to set the default panel when the editor is reopened. Missing a dedicated event for 'opening'.
```
Comment (adiux) [inline src/buttons.js:155]: I think this part is not needed anymore when we dont keep the history. I will remove it and create a new PR

**Pull request merged**